apps/survey/contracts/Surveying.sol

```diff
@@ -0,0 +1,240 @@
+    }
+
+    Survey[] surveys;
+
+    event StartSurvey(uint256 indexed surveyId);
+    event CastVote(uint256 indexed surveyId, address indexed voter, uint256 option, uint256 stake);
+    event ChangeMinParticipation(uint256 minAcceptParticipationPct);
+
+    /**
+    * @notice Initializes Survey app with `_token.symbol(): string` for governance, minimum acceptance participation of `(_minAcceptParticipationPct - _minAcceptParticipationPct % 10^14) / 10^16` and durations of `(_surveyTime - _surveyTime % 86400) / 86400` day `_surveyTime >= 172800 ? 's' : ''`
+    * @param _token MiniMeToken address that will be used as governance token
+    * @param _minAcceptParticipationPct Percentage of total voting power that must participate in a survey for it to be taken into account (expressed as a 10^18 percentage, (eg 10^16 = 1%, 10^18 = 100%)
+    * @param _surveyTime Seconds that a survey will be open for token holders to vote
+    */
+    function initialize(
+        MiniMeToken _token,
+        uint256 _minAcceptParticipationPct,
+        uint64 _surveyTime
+    ) onlyInit external
+    {
+        initialized();
+
+        require(_minAcceptParticipationPct > 0 && _minAcceptParticipationPct <= PCT_BASE);
+
+        token = _token;
+        minAcceptParticipationPct = _minAcceptParticipationPct;
+        surveyTime = _surveyTime;
+    }
+
+    /**
+    * @notice Change minimum acceptance participation to `(_minAcceptParticipationPct - _minAcceptParticipationPct % 10^14) / 10^16`%
+    * @param _minAcceptParticipationPct New acceptance participation
+    */
+    function changeMinAcceptParticipationPct(uint256 _minAcceptParticipationPct) authP(MODIFY_PARTICIPATION_ROLE, arr(_minAcceptParticipationPct)) external {
+        require(_minAcceptParticipationPct > 0);
+        require(_minAcceptParticipationPct <= PCT_BASE);
+        minAcceptParticipationPct = _minAcceptParticipationPct;
+
+        ChangeMinParticipation(_minAcceptParticipationPct);
+    }
+
+    /**
+    * @notice Create a new non-binding survey about "`_metadata`"
+    * @param _metadata Survey metadata
+    * @param _options Number of options voters can decide between
+    * @return surveyId id for newly created survey
+    */
+    function newSurvey(string _metadata, uint256 _options) auth(CREATE_SURVEYS_ROLE) external returns (uint256 surveyId) {
+        return _newSurvey(_metadata, _options);
+    }
+
+    /**
+    * @notice Vote for multiple options in survey #`_surveyId`.
+    * @dev Empty options and stakes arrays allow to remove previous votes
+    * @param _surveyId Id for survey
+    * @param _options Array with indexes of supported options
+    * @param _stakes Number of tokens assigned to each option
+    */
+    function voteOptions(uint256 _surveyId, uint256[] _options, uint256[] _stakes) public {
+        require(_options.length == _stakes.length);
+        require(canVote(_surveyId, msg.sender));
+
+        uint256 i;
```

Comment:
Because it's being used twice, to avoid using 2 variables. Doing `for (uint256 i =..` in each for loop was throwing an error (`DeclarationError: Identifier already declared`).

Reply:
You can name it differently, `for (uint j = 1; ...)`. There is no benefit in just using one variable

Reply:
there is no issue with that and no benefit in reusing the value. it is just confusing, you can rename the second for loop counter to `j`